src/main/java/io/vlingo/xoom/lattice/grid/application/GridApplicationMessageHandler.java

```diff
@@ -69,26 +74,52 @@ public GridApplicationMessageHandler(
     this.visitor = new ControlMessageVisitor();
   }
 
+  @Override
+  public void informNodeIsHealthy(final Id id, final boolean isHealthy) {
+    isClusterHealthy.set(isHealthy);
+    if (isHealthy) {
+      disburse(id);
+    }
+  }
+
   @Override
   public void handle(final RawMessage raw) {
     try {
       final Message message = decoder.decode(raw.asBinaryMessage());
       final Id sender = Id.of(raw.header().nodeId());
+      logger.debug("Buffering message {} from {}", message, sender);
       final Runnable runnable = () -> {
         logger.debug("Handling message {} from {}", message, sender);
         message.accept(localNode, sender, visitor);
       };
 
+      if (isClusterHealthy.get()) {
+        runnable.run(); // incoming messages are dispatched immediately
+      } else {
+        buffer.offer(runnable); // buffer messages; cluster is not healthy
```

Comment:
@pflueras Based on your answer in the PR merge if there is no concern. I was not warning as much a wondering. So I am approving, but please consider and reply. Thanks!

Reply:
Yes, incoming messages on a node of an unhealthy cluster are buffered mainly because the answer might not be delivered at all. Also I am not sure if processing of messages is an idempotent operation. I was trying to minimize double processing of the same message especially for the cases when this operation is expensive.
If throughput of a cluster in turbulent conditions is affected by this cautious approach I do agree to disable it.
I would propose to keep it for now.